kedro/contrib/idempotent/idempotent_runner.py

```diff
@@ -40,29 +41,69 @@
 from kedro.runner.runner import AbstractRunner
 
 
-def run_node_idempotently(node: Node, catalog: DataCatalog, state: IdempotentStateStorage) -> Node:
+def get_hash_value(data):
```

Comment:
Move this function into the `generate_run_id`.

Reply:
That way, we will need to determine if the data is parameter/MemoryDataset inside of  `generate_run_id`.

Reply:
We don't need to determine if the data is a parameter/MemoryDataSet inside of `generate_run_id`, because we never knew if the data was a parameter or MemoryDataSet inside of the `get_hash_value` function.

Reply:
"we never knew if the data was a parameter or MemoryDataSet inside of the get_hash_value function" that's half true, because `get_hash_value` is called conditionally.
But yes we can use the catalog's check on MemoryDataSet. I will work on that later, after we pass `catalog` in `IdempotentStateStorage`